### PR TITLE
Freq avg prop flags

### PR DESCRIPTION
**Pull request by mwilensky768**
This change adds an optional keyword to `UVData.frequency_average` called `propagate_flags` which affects the resulting flags during the averaging process.

## Description

The default setting for `propagate_flags` is `False`. When `propagate_flags` is set to `True`, the resulting flags are calculated according to the following rule. If _any_ samples that contribute to an average sample are flagged, the corresponding entry in the new flag array will be set to `True`. When `False`, a resulting flag is only `True` if all contributing samples to the average sample were flagged to begin with, which is the current behavior. In either case, the resulting data value is calculated according to the current behavior, which is that only unflagged samples contribute to the average sample, except when all contributing samples are flagged, in which case all samples contribute to the average. This is accomplished with two if statements.

## Motivation and Context

Spectral differences in flagging and subsequent averaging can lead to spectral fluctuations in the data due to unequal weighting, which can cause problems in certain radio applications e.g. [Offringa et al., 2019](https://arxiv.org/abs/1901.04752). It is convenient to have such "partially flagged" samples optionally marked as flagged during the averaging process.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation change (documentation changes only)
- [ ] Version change
- [ ] Build or continuous integration change

## Checklist:

- [x] I have read the [contribution guide](https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/master/.github/CONTRIBUTING.md).
- [x] My code follows the code style of this project.

Bug fix checklist:
- [ ] My fix includes a new test that breaks as a result of the bug (if possible).
- [ ] All new and existing tests pass.
- [ ] I have updated the [CHANGELOG](https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/master/CHANGELOG.md).

New feature checklist:
- [x] I have added or updated the docstrings associated with my feature using the [numpy docstring format](https://numpydoc.readthedocs.io/en/latest/format.html).
- [ ] I have updated the tutorial to highlight my new feature (if appropriate).
- [x] I have added tests to cover my new feature.
- [x] All new and existing tests pass.
- [x] I have updated the [CHANGELOG](https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/master/CHANGELOG.md).

Breaking change checklist:
- [ ] I have updated the docstrings associated with my change using the [numpy docstring format](https://numpydoc.readthedocs.io/en/latest/format.html).
- [ ] I have updated the tutorial to reflect my changes (if appropriate).
- [ ] My change includes backwards compatibility and deprecation warnings (if possible).
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests pass.
- [ ] I have updated the [CHANGELOG](https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/master/CHANGELOG.md).

Documentation change checklist:
- [ ] Any updated docstrings use the [numpy docstring format](https://numpydoc.readthedocs.io/en/latest/format.html).
- [ ] If this is a significant change to the readme or other docs, I have checked that they are rendered properly on ReadTheDocs. (you may need help to get this branch to build on RTD, just ask!)

Version change checklist:
- [ ] I have updated the [CHANGELOG](https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/master/CHANGELOG.md) to put all the unreleased changes under the new version (leaving the unreleased section empty).
- [ ] I have noted any dependency changes since the last version and will update the conda package build accordingly.

Build or continuous integration change checklist:
- [ ] If required or optional dependencies have changed (including version numbers), I have updated the readme to reflect this.
- [ ] If this is a new CI setup, I have added the associated badge to the readme and to references/make_index.py (if appropriate).

### CODE REVIEW

**Comment by mwilensky768** [comment]
Unfortunately I committed before installing pre-commit, but I don't think I violated the style anywhere.

**Comment by bhazelton** [comment]
It's not formatted the way Black wants it to be. You can fix that by running the pre-commit hook on the command line and then committing the changes that Black makes. To run it on the command line run `pre-commit run -a`.

It looks like you also need to rebase to fix the changelog, so maybe do that first and then commit and push the Black fixes.

**Comment by mwilensky768** [comment]
Rebased and reformatted.

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,9 @@ All notable changes to this project will be documented in this file.
 
 ## [Unreleased]
 
+### Added
+- `propagate_flags` keyword for `UVData.frequency_average` for different resulting flags from averaging
```
Comment (bhazelton) [inline CHANGELOG.md:7]: Can you make this log message a little clearer in terms of what the new functionality is?

**Pull request merged**